src/hackney_pool.erl

```diff
@@ -422,7 +409,7 @@ dequeue(Dest, Ref, State) ->
     empty ->
       State#state{clients = Clients2};
     {ok, {From, Ref2, _Requester}, Queues2} ->
-      Pending2 = del_pending(Ref, Pending),
+      Pending2 = del_pending(Ref2, Pending),
```

Comment:
This was a leak: deleting pending by `Ref` made no sense, as it is a Reference to "active" client (the one which is in '#state.clients' dict.  It lead to situation when queue was emptied, but pending dict was still keeping outdated clients.